docs: Instructions for teleoperation added (#108)
ROAR-QUTRC/perseus-v2: docs/source/home/teleoperation.md

# Perseus Teleoperation

This guide covers teleoperation of Perseus using a keyboard or Xbox controller. Teleoperation enables remote control of Perseus's movement. Operation of payloads (bucket, arms) is not covered in this guide.

## Prerequisites

The following conditions must be met before beginning:

- The laptop runs a Linux distribution with the latest code from the perseus-v2 Github repository:
  ```console
  git checkout main
  git pull
  ```

> - Required fonts are installed for rendering the Orin's zsh shell with power10k theme (available in /perseus-v2/fonts)

Reviewer: ```suggestion
- Required fonts are installed for rendering the Orin's zsh shell with the [`powerlevel10k`](https://github.com/romkatv/powerlevel10k) theme (available in /perseus-v2/fonts). You will likely have to set the font to be your terminal's default (it'll come up as `MesloLGS NF`), though you may need to restart your terminal for the font to be available after installation.
```

2. Connect the battery's XT90 connector to the power meter.

### Network Verification

Perseus should automatically connect to the ROAR-QUTRC network. Verify the connection using one of these methods:

- Check the Unifi console for connected devices
- Use arp-scan on the laptop to list devices with allocated IP addresses:
  ```console
  sudo arp-scan --localnet
  ```

#### Network Troubleshooting

1. Physical Connectivity

   - Confirm Unifi UX is powered and fully booted (screen shows device count)
   - Verify power LED indicators are lit
   - Check E-Stop position as it affects system power

2. Laptop Connectivity

   - Verify connection to "ROAR-QUTRC" network
   - Check signal strength (maintain line of sight with Unifi UX if possible)
   - Confirm IP address allocation using `ip addr`

3. Perseus Connectivity

   ```console
   # Verify Perseus visibility
   ping big-brain.local

   # Alternative: network device scan
   sudo arp-scan --localnet | grep -i nvidia
   ```

### System Access

Access Perseus via SSH from the laptop:

```console
ssh qutrc@big-brain.local
```

### Perseus Software Launch

Execute the following command on Perseus:

```console

> nix run .#perseus

Reviewer: Probably worth noting that internet access is required for this if you can't guarantee it already has everything it needs downloaded from a previous run.

```

### Control Software Launch

Launch the appropriate control software on the laptop based on the desired control method.

#### Keyboard Control

> ```console
> cd perseus-v2/software/ros_ws
> colcon build
> source install/setup.bash
> ros2 run teleop_twist_keyboard teleop_twist_keyboard
> ```

Reviewer: This won't work, because we explicitly split the domain IDs. Either none or both of them need to be run from the production environment.
````suggestion
```console
cd perseus-v2
nix run .#ros2 -- run teleop_twist_keyboard teleop_twist_keyboard
```
````

> Note: Control messages are only sent when the terminal has focus and receives keystrokes.

Reviewer: Last time we tested, it looked like there wasn't a timeout happening to stop the motors (even though `diff_drive_controller` should enforce one). Not sure why that was happening. Probably worth investigating.
```suggestion
Note: Control messages are only sent when the terminal has focus and receives keystrokes ({keys}`k` to stop).
```

#### Xbox Controller

> ```console
> cd perseus-v2/software/ros_ws
> colcon build
> source install/setup.bash
> ros2 run input_devices xbox_controller --ros-args --remap use_stamped_message:=true
> ```

Reviewer: Same thing.
````suggestion
```console
cd perseus-v2
nix run .#ros2 -- run input_devices xbox_controller --ros-args -p use_stamped_message:=true
```
````
I'm guessing that `--remap` is meant to be a parameter?

### Safe Operation

#### Safety Protocol

Before operating Perseus:

1. Ensure all bystanders:
   - Are aware of Perseus's operation
   - Know the E-Stop button location
   - Understand they should press the E-Stop if they observe:
     - Collision risks
     - Unsafe behaviour
     - Control issues
     - Any hazardous situations

> The E-Stop, a red mushroom-shaped button on Perseus's top, immediately cuts drive motor power when pressed.

Reviewer: ```suggestion
The E-Stop, a red mushroom-shaped button on Perseus's top, immediately cuts all power when pressed.
```
